### ISSUE TITLE
doctor: unknown workspace and cache directory on some windows mapped network shares

### ISSUE THREAD

**Issue by anasitomtn** [iterative/dvc#6305]
# Bug Report

doctor: unknown workspace and cache directory on some windows mapped network shares

## Description

Hi everyone !
I came across the following error: https://dvc.org/doc/user-guide/troubleshooting#no-dvc-cache
, except that I have set the path to my cache folder correctly, nevertheless it is not recognized in "dvc doctor".
I think it's due to the configuration I'm using: I'm on a shared network drive mounted on Windows 10.
Maybe it is linked to this issue : https://github.com/iterative/dvc/issues/5832

### Reproduce

1. Shared NAS on Windows 10 (N:/)
2. "$ dvc doctor" in a DVC repo with cache configured, inside our N:/
3. Result :
```
DVC version: 2.5.4 (pip)
Platform: Python 3.9.1 on Windows-10-10.0.17763-SP0
Supports:
        http (requests = 2.25.0),
        https (requests = 2.25.0)
Cache types:
Cache directory: ('unknown', 'none')
Caches: local
Remotes: None
Workspace directory: ('unknown', 'none')
Repo: dvc, git
```

### Expected

Expected something like this :
Cache directory: NTFS on N:\
Caches: local
Workspace directory: NTFS on N:\

Thank you! 
:)

**Comment by efiop** [comment]
Hi @AnasChetouan !

> I came across the following error: https://dvc.org/doc/user-guide/troubleshooting#no-dvc-cache

Could you show the error you are getting from a dvc command, please? E.g. `dvc status -v`

Could you show `pip freeze | grep psutil`, please?

**Comment by AnasChetouan** [comment]
Hi @efiop, thanks for your reply!

> Could you show `pip freeze | grep psutil`, please?
psutil==5.8.0

> I came across the following error: https://dvc.org/doc/user-guide/troubleshooting#no-dvc-cache
Sorry, I may have explained it wrong. The error is that it's printing Cache directory: ('unknown', 'none') instead of the partition type.

I have found a trail that can lead to the solution of the problem : 
I've done some tests, using code from https://github.com/iterative/dvc/blob/master/dvc/info.py and I've noticed that on line 148, the problem appears after using "Path.resolve()".

Path before resolve : N:/Projects/XX/XX/...
Path after resolve : \\<nas-name>\dfs_root\Projects\XX\XX\...

In this way, we're not able to match our partition name on line 142 because we're not using the "N:/" prefix anymore but the NAS name directly. I think that's why we're not able to get the file system type.

I can propose you to look into it further on my side and why not bring a fix if I can?

**Comment by efiop** [comment]
@AnasChetouan Ah, so you are not experiencing any other issues with dvc, right? Just making sure :slightly_smiling_face: 

Thanks for the investigation, makes total sense! Here is a quick explanation: https://stackoverflow.com/questions/38325636/why-does-pythons-pathlib-and-os-lib-return-different-results-for-mapped-network 

> On Windows, resolve calls os.path._getfinalpathname, which calls GetFinalPathNameByHandle. This starts from the final resolved path of the file in the NT namespace (e.g. \Device\Mup\server\share\...) and works backward as required to get the VOLUME_NAME_DOS. If the final path is a UNC share (i.e. \Device\Mup), it returns a path of the form \\?\UNC\server\share\.... pathlib translates this back to a regular UNC path of the form \\server\share\....

Had no idea resolve() behaved like that on windows :man_shrugging: So seems like we just need to also `resolve()` `partition` keys and that should fix it. Would you be interested in sending a fix? :slightly_smiling_face:

**Comment by AnasChetouan** [comment]
> @AnasChetouan Ah, so you are not experiencing any other issues with dvc, right? Just making sure 🙂

For the moment I have not encountered any other problems but I will not hesitate to report this to you in the future if necessary
🙂

> Had no idea resolve() behaved like that on windows 🤷‍♂️ So seems like we just need to also `resolve()` `partition` keys and that should fix it. Would you be interested in sending a fix? 🙂

Yes with pleasure ! Here is my fix proposal.

Thank you. :)

**Comment by anasitomtn** [comment]
I've just noticed that "Cache types:" is also not working, I think it's related to this path problem.
I'll check this further on the __get_linktype_support_info_ function and get back to you soon :)

**Comment by efiop** [comment]
@anasitomtn Thank you! Let us know if you'll have any questions.

**Comment by anasitomtn** [comment]
Yes, of course, I will investigate further to see if this path problem broke something else, especially with symbolic links and will do my best to bring a fix soon, with some testing as well.

Have a nice day and see you soon!

**Comment by anasitomtn** [comment]
Hello ! I'm back with news, I'm really sorry for the delay.

I was able to take a closer look at the 2nd problem which is at the level of the cache types. Indeed, I noticed that for my hardlinks on my NAS server on Windows, my links are in "broken" status (in dvc/dvc/info.py:__get_linktype_support_info) and after several researches, it seems to me that the problem comes from my own configuration. Here is a link that corresponds to my situation: https://superuser.com/questions/1017653/creating-symbolic-links-on-networked-drive

I think it's a problem I don't have the power to act on since I don't have administrator rights on my company's shared NAS, so I'm unable to go further on the investigation of the hardlinks creation failure for the moment.

So it's not a DVC problem on this side, everything seems fine, it's just that the os.link() command doesn't work at all on my NAS and so I'm waiting for a solution on this side to have symbolic links of my data files, if you have a solution for that, I'm interested!

Finally, the basic resolution which consisted in setting the display of the blocked cache and workspace folders to "unknown" should be fixed with the pull request I sent you. Don't hesitate to tell me if there is something to fix/add!

Thank you and see you soon ! 🙂
